### PR TITLE
feat: :sparkles: sort exercise lists by list name

### PR DESCRIPTION
**Pull request by mike011**
## Description

The exercise list can now be sorted alphabetically. This has been accomplished by creating a new `ExerciseListContainerView` which contains the `ExerciseListView` to be sorted. From the menu when you pick `Alphabetical` it sets the `$sortOrder` to being a new `SortDescriptor` using the `ExerciseList.name`.  

This work was heavily influenced by: [Sorting query results by Paul Hudson](https://www.hackingwithswift.com/quick-start/swiftdata/sorting-query-results)

## How Has This Been Tested?
By default when you see the view it’s sorted alphabetically, and when you add something new it add’s it in, in the correct location. So I hacked up reverse alphabetical button, then verified pressing alphabetical correctly sorts the list.

## Screenshots (if appropriate):

https://github.com/user-attachments/assets/8ceb8b7c-bfa6-4b4b-ba7f-19aabdbc2106

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Developer tooling (fix or improve developer tooling)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

**Comment by mike011** [comment]
I don’t understand why the check is failing.

**Comment by mike011** [comment]
@heyjaywilson pls review

File: AppPackages/Exercises/Sources/Exercises/ExerciseLists/ExerciseListView.swift [hunk 1]
```diff
@@ -4,68 +4,38 @@
...
-public struct ExerciseListView: View {
-	@Environment(\.modelContext) private var modelContext
+struct ExerciseListView: View {
+    @Environment(\.modelContext) private var modelContext
+    @Query private var lists: [ExerciseList]
 
-	// Sorting by name here because I don't have an order field
-	@Query(sort: \ExerciseList.name, order: .forward, animation: .default) private var lists:
-		[ExerciseList]
+    init(sortOrder: SortDescriptor<ExerciseList>) {
```
Comment (heyjaywilson) [inline AppPackages/Exercises/Sources/Exercises/ExerciseLists/ExerciseListView.swift:18]: good call here on passing it in and making the query dynamic

**Review by heyjaywilson — APPROVED** [review]
Thanks for the work!

**Pull request merged**